src/Xamarin.Android.Build.Tasks/Tasks/AndroidTask.cs

```diff
@@ -34,12 +34,15 @@ public abstract class AndroidAsyncTask : AsyncTask
 
 		public abstract string TaskPrefix { get; }
 
+		[Obsolete("You should not use the 'Log' property directly for AsyncTask. Use the 'Log*' methods instead.", error: true)]
+		public new TaskLoggingHelper Log { get; }
```

Comment:
I realize it has `error:true`, but should this also throw an exception for Really Good Measure?

```csharp
Obsolete (..., error: true)]
public new TaskLoggingHelpe Log {
    get => throw new NotSupportedException ("Should not be reached!");
}
```

Admittedly, if the new `Log` property is accessed e.g. via Reflection, it'll "just" return `null` (and possibly NRE?), and it really shouldn't happen at all *anyway*, but...  defense in depth?  ¯\_(ツ)_/¯

Reply:
I would be careful about throwing an exception. the base class AsyncTask uses `Log` allot when on the UI thread. See https://github.com/xamarin/Xamarin.Build.AsyncTask/blob/master/Xamarin.Build.AsyncTask/AsyncTask.cs#L138 note all the #pragma stuff which ignore that Obsolete warning.

Reply:
Because I used the `new` keyword, the base class shouldn't be able to call the property. I think throwing might be better than the NRE, as I could put a better exception message.

Reply:
Or maybe I should just return `base.Log`? Might be the absolute safest option.